src/test/java/StorageTest.java

```diff
@@ -0,0 +1,24 @@
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class StorageTest {
+    @Test
+    public void testWriteFile() {
+        Ui ui = new Ui();
+        Storage storage = new Storage("data/duke.txt", ui);
+        TaskList taskList = new TaskList(new ArrayList<>());
+        Task task = new Todo("read book");
+        taskList.arr.add(new Todo("read book"));
+        storage.writeFile(taskList);
+        try {
+            BufferedReader br = new BufferedReader(new FileReader("data/duke.txt"));
+            assertEquals("T/0/read book", br.readLine());
+        } catch (Exception ignored) {
+        }
```

Comment:
Perhaps can add `fail()` in the `catch` case here. If the file cannot be found despite writing to it, then something may be wrong. The `Storage` class should ideally generate the file at the correct location before writing to it. If there are conditions for the file to be created, perhaps you can split this test case into two separate test cases - writing to a non-existent file that won't be created at all, and writing to a file that will be created in the process.